core/src/components/picker-internal/picker-internal.tsx

```diff
@@ -0,0 +1,503 @@
+import { Component, ComponentInterface, Element, Event, EventEmitter, Host, h } from '@stencil/core';
+
+import { getElementRoot } from '../../utils/helpers';
+
+import { PickerInternalChangeEventDetail } from './picker-internal-interfaces';
+
+/**
+ * @virtualProp {"ios" | "md"} mode - The mode determines which platform styles to use.
+ * @internal
+ */
+@Component({
+  tag: 'ion-picker-internal',
+  styleUrls: {
+    ios: 'picker-internal.ios.scss',
+    md: 'picker-internal.md.scss'
+  },
+  shadow: true
+})
+export class PickerInternal implements ComponentInterface {
+  private inputEl?: HTMLInputElement;
+  private inputMode = false;
+  private inputModeColumn?: HTMLIonPickerColumnInternalElement;
+  private highlightEl?: HTMLElement;
+  private actionOnClick?: () => void;
+  private destroyKeypressListener?: () => void;
+
+  @Element() el!: HTMLIonPickerInternalElement;
+
+  @Event() ionInputModeChange!: EventEmitter<PickerInternalChangeEventDetail>;
+
+  componentWillLoad() {
+    getElementRoot(this.el).addEventListener('focusin', this.onFocusIn);
+    getElementRoot(this.el).addEventListener('focusout', this.onFocusOut);
```

Comment:
`@Listen` only lets you listen on the `body`, `document`, or `window`, so if we used Listen it would pick up all focusin/focusout events.

Setting the events on `Host` sets the event listeners on `<ion-picker-internal>` whereas the focusin/focusout events should be set on the `#shadowRoot` which is one level deeper. (getElementRoot returns an element's shadow root if there is one).